plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -601,7 +601,8 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
         // This method does not calculate column metadata for the projected columns
         checkArgument(tableHandle.getProjectedColumns().isEmpty(), "Unexpected projected columns");
         Table icebergTable = catalog.loadTable(session, tableHandle.getSchemaTableName());
-        List<ColumnMetadata> columns = getColumnMetadatas(SchemaParser.fromJson(tableHandle.getTableSchemaJson()));
+        // Could also use tableHandle.getTableSchemaJson but icebergTable is loaded anyway
+        List<ColumnMetadata> columns = getColumnMetadatas(icebergTable.schema());
```

Comment:
This is actually wrong. `SchemaParser.fromJson(tableHandle.getTableSchemaJson())` contains a different (and better) schema.

Reply:
(when reading from a version)

Reply:
Did any tests fail? Is it worth leaving a code comment?

Reply:
yes, it was caught in dedicated tests, i think we're good